src/svmlin.cpp

```diff
@@ -0,0 +1,129 @@
+
+  NumericVector ycop( y.begin(), y.end() );
+  NumericVector costcop( costs.begin(), costs.end() );
+//   Rprintf("Step 1\n");
+//   size_t size = ((DoubleVector)X.slot("x")).length()+((DoubleVector)Xu.slot("x")).length();
+//   std::vector<double> vals(size);
+//   std::vector<double>::iterator it = vals.begin();
+//   vals.insert(it,((DoubleVector)X.slot("x")).begin(),((DoubleVector)Xu.slot("x")).end());
+//   it = vals.begin()+((DoubleVector)X.slot("x")).length();
+//   vals.insert(it,((DoubleVector)Xu.slot("x")).begin(),((DoubleVector)Xu.slot("x")).end());
+//
+//   Rprintf("Step 2\n");
+//
+//   size = ((IntegerVector)X.slot("i")).length()+((IntegerVector)Xu.slot("i")).length();
+//   std::vector<int> colinds(size);
+//   std::vector<int>::iterator it2 = colinds.begin();
+//   colinds.insert(it2,((IntegerVector)X.slot("i")).begin(),((IntegerVector)X.slot("i")).end());
+//   it2 = colinds.begin() + ((IntegerVector)X.slot("i")).length();
+//   colinds.insert(it2,((IntegerVector)Xu.slot("i")).begin(),((IntegerVector)Xu.slot("i")).end());
+//
+//   size = ((IntegerVector)X.slot("p")).length()+((IntegerVector)Xu.slot("p")).length();
+//   std::vector<int> rowpts(size);
+//   it2 = rowpts.begin();
+//   rowpts.insert(it2,((IntegerVector)X.slot("p")).begin(),((IntegerVector)X.slot("p")).end());
+//   it2 = rowpts.begin() + ((IntegerVector)X.slot("p")).length();
+//   rowpts.insert(it2,((IntegerVector)Xu.slot("p")).begin(),((IntegerVector)Xu.slot("p")).end());
+
+  // R data to svmlin data structure
+  Data->m=((IntegerVector)X.slot("Dim"))[1];
+  Data->l=l;
+  Data->u=Data->m-Data->l;
+  Data->n=((IntegerVector)X.slot("Dim"))[0];
+  Data->nz=((DoubleVector)X.slot("x")).size();
+  Data->val=((DoubleVector)X.slot("x")).begin();
+  Data->rowptr=((IntegerVector)X.slot("p")).begin();
+  Data->colind=((IntegerVector)X.slot("i")).begin();
+  Data->Y=ycop.begin();
+  Data->C=costcop.begin();
+
+  // TODO: load correct costs for unlabeled data.
+  if (Options->verbose) {
+    Rcout << "  Input Data Matrix Statistics:" << endl;
+    Rcout << "      Examples: " << Data->m << endl;
+    Rcout << "      Features: " << Data->n << " (including bias feature)" << endl;
+    Rcout << "      Non-zeros:  " << Data->nz << " (including bias features)" << endl;
+    Rcout << "      Average sparsity: " << Data->nz*1.0/Data->m << " non-zero features per example." << endl;
+  }
+    //   for (int i = 0; i<((DoubleVector)X.slot("x")).length();i++) {
+//     Rprintf("val: %f \n",Data->val[i]);
+//   }
+//   for (int i = 0; i<((IntegerVector)X.slot("i")).length();i++) {
+//     Rprintf("col: %d \n",Data->colind[i]);
+//   }
+//   for (int i = 0; i<((IntegerVector)X.slot("p")).length();i++) {
+//     Rprintf("row: %d \n",Data->rowptr[i]);
+//   }
+
+
+  // Run
+  ssl_train(Data,
+            Options,
+            Weights,
+            Outputs);
+//Clear(Data);
+
+  return Rcpp::List::create(Rcpp::Named("Weights") = std::vector<double>(Weights->vec, Weights->vec+Weights->d),
```

Comment:
The elements of `Rcpp::List` should be  `Rcpp::NumericVector` instead of `std::vector<double>`.

Reply:
changed.